indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,271 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from typing import Optional
+
+from flask import flash
+from google.auth import crypt, jwt
+from google.auth.transport.requests import AuthorizedSession
+from google.oauth2.service_account import Credentials
+from requests.exceptions import RequestException
+from werkzeug.exceptions import BadRequest, ServiceUnavailable
+
+from indico.core import signals
+from indico.core.config import config
+from indico.core.logger import Logger
+from indico.util.i18n import _
+
+
+class GoogleWalletManager:
+    """Google Wallet management class."""
+
+    def __init__(self, event, registration=None):
+        self.event = event
+        self.registration = registration
+        self.settings = self.get_google_wallet_settings(self.event.category)
+        self.credentials = None
+        self.http_client = None
+        self.base_url = 'https://walletobjects.googleapis.com/walletobjects/v1'
+        self.batch_url = 'https://walletobjects.googleapis.com/batch'
+        self.class_url = f'{self.base_url}/eventTicketClass'
+        self.object_url = f'{self.base_url}/eventTicketObject'
+
+        # Set up authenticated client
+        if self.settings:
+            self.auth()
+
+    @staticmethod
+    def get_google_wallet_settings(category):
+        while category:
+            if category.google_wallet_settings:
+                return category.google_wallet_settings
+            else:
+                category = category.parent
+        return None
+
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        self.credentials = Credentials.from_service_account_info(
+            self.settings['google_wallet_application_credentials'],
+            scopes=['https://www.googleapis.com/auth/wallet_object.issuer'])
+        self.http_client = AuthorizedSession(self.credentials)
+
+    def create_class_template(self, issuer_id: str, class_suffix: str) -> dict:
+        """This method will return a dict format ticket class template."""
+        from indico.modules.categories.controllers.util import make_format_event_date_func
+        logo_url = config.ABSOLUTE_LOGO_URL if config.LOGO_URL else \
+            f'{config.BASE_URL}{config.IMAGES_BASE_URL}/logo_indico.png'
```

Comment:
```suggestion
        logo_url = config.ABSOLUTE_LOGO_URL or urljoin(config.BASE_URL, f'{config.IMAGES_BASE_URL}/logo_indico.png')
```

(you need to add the import for urljoin of course, and this if untested but should work)

Reply:
PS: We generally avoid `\` wrapping in favor of `(...)`

Reply:
That's my pycharm default action...

Reply:
Maybe there's a setting to avoid that? I haven't used PyCharm for a few years, so not sure...